static/app/views/dashboardsV2/dashboard.tsx

```diff
@@ -264,37 +265,39 @@ class Dashboard extends Component<Props, State> {
   };
 
   handleUpdateComplete = (prevWidget: Widget) => (nextWidget: Widget) => {
-    const {isEditing, handleUpdateWidgetList} = this.props;
-    const nextList = [...this.props.dashboard.widgets];
+    const {isEditing, onUpdate, handleUpdateWidgetList} = this.props;
+
+    let nextList = [...this.props.dashboard.widgets];
     const updateIndex = nextList.indexOf(prevWidget);
-    nextList[updateIndex] = {...nextWidget, tempId: prevWidget.tempId};
-    this.props.onUpdate(nextList);
+    const nextWidgetData = {
+      ...nextWidget,
+      tempId: prevWidget.tempId,
+    };
+
+    // Only modify and re-compact if the default height has changed
+    if (
+      getDefaultWidgetHeight(prevWidget.displayType) !==
+      getDefaultWidgetHeight(nextWidget.displayType)
+    ) {
+      nextList[updateIndex] = enforceWidgetHeightValues(nextWidgetData);
+      nextList = generateWidgetsAfterCompaction(nextList);
```

Comment:
Just a nit suspicion, but I have a feeling you don't actually need to call compaction here since it looks to me that `enforceWidgetHeightValues` will always return at least the previous height on existing widgets, so it's not possible for a widget to get smaller ie no gaps are created therefore no compaction needed.

Not an issue though, just an observation

Reply:
Ah, I think the term `compact` is a bit ambiguous here. It also resolves collisions and that's why we're calling it here, to update the other widgets that would be impacted by the minimum height forcing a widget to take up 2 rows

Should be called something like `resolveLayout` 🤷

Reply:
Should I rename `generateWidgetsAfterCompaction` to something like `assignResolvedLayoutToWidgets`?

Reply:
Ah true, I think we talked about this before lol. I think `generateWidgetsAfterCompaction` is probably fine, since `compact` is the term used by RGL

Reply:
I'll keep it for now